src/opCodeFinder.js

```diff
@@ -0,0 +1,178 @@
+import Web3 from "web3";
+import fs from "fs";
+
+let web3;
+let opCode;
+let matchContractCount = 0;
+
+async function validateInputs(args) {
+  let startBlock, endBlock, opcode, rpcUrl;
+  let validArgs = [];
+  // Check if all parameters are provided
+  if (args.length !== 4) {
+    console.log(
+      "Please provide exactly four arguments: startBlock, endBlock, opcode, and rpc url."
+    );
+    process.exit(1);
+  }
+
+  rpcUrl = args[3];
+  let latestBlock;
+  try {
+    web3 = new Web3(new Web3.providers.HttpProvider(rpcUrl));
+    latestBlock = await web3.eth.getBlockNumber();
+  } catch (e) {
+    console.log("rpc url must be a valid URL.");
+    process.exit(1);
+  }
+
+  // Get the parameter values
+  startBlock = parseInt(args[0], 10);
+  if (args[1] === "latest") {
+    endBlock = Number(latestBlock);
+    console.log("latest Block Number is " + endBlock);
+  } else {
+    endBlock = parseInt(args[1], 10);
+  }
+  opcode = parseInt(args[2], 16);
+
+  // Validate that the block number is a valid positive integer
+  if (
+    isNaN(startBlock) ||
+    isNaN(endBlock) ||
+    startBlock < 0 ||
+    endBlock < 0 ||
+    startBlock > endBlock ||
+    startBlock > latestBlock ||
+    endBlock > latestBlock
+  ) {
+    console.log("Invalid block number!");
+    process.exit(1);
+  }
+
+  // Validate that the opcode is a valid hexadecimal string (optionally starting with "0x")
+  if (!/^0x[a-fA-F0-9]+$/i.test(opcode) && !/^[a-fA-F0-9]+$/i.test(opcode)) {
+    console.log(
+      "Opcode must be a valid hexadecimal string, optionally starting with '0x'."
+    );
+    process.exit(1);
+  }
+
+  // If all validations pass
+  validArgs.push(startBlock, endBlock, opcode, rpcUrl);
+  return validArgs;
+}
+
+async function findContracts(startBlockNumber, endBlockNumber) {
+  let contractCount = 0;
+
+  for (let i = startBlockNumber; i <= endBlockNumber; i++) {
+    process.stdout.write(
+      `\rSearching for block ${i} ,total found ${contractCount} contracts, ${matchContractCount} of which meet the criteria`
+    );
+    let block = await web3.eth.getBlock(i);
+    if (block != null) {
+      if (block.transactions != null) {
+        let blockFinishedCount = block.transactions.length;
+        //search for type of "CREATE" in each transaction
+        for (let txHash of block.transactions) {
+          let traceObject = await web3.eth.currentProvider.sendAsync({
+            method: "debug_traceTransaction",
+            params: [`${txHash}`, { tracer: "callTracer" }],
+            jsonrpc: "2.0",
+            id: "1",
+          });
+          if (traceObject.result.type === "CREATE") {
```

Comment:
You're on the right path. This part needs some modification: namely we should check also the nested calls within the trace and collect all of the CREATEs.